src/containerapp/azext_containerapp/containerapp_resiliency_decorator.py

```diff
@@ -0,0 +1,476 @@
+                }
+            http_retry_policy['matches'] = matches
+            containerapp_def['properties']['httpRetryPolicy'] = http_retry_policy
+
+        # Timeouts
+        timeout_policy = safe_get(containerapp_def, 'properties', 'timeoutPolicy')
+        if timeout_policy and ('responseTimeoutInSeconds' in timeout_policy or 'connectionTimeoutInSeconds' in timeout_policy):
+            timeout_policy['responseTimeoutInSeconds'] = timeout_policy.get('responseTimeoutInSeconds', DEFAULT_RESPONSE_TIMEOUT)
+            timeout_policy['connectionTimeoutInSeconds'] = timeout_policy.get('connectionTimeoutInSeconds', DEFAULT_CONNECTION_TIMEOUT)
+            containerapp_def['properties']['timeoutPolicy'] = timeout_policy
+
+        # Circuit Breaker
+        circuit_breaker_policy = safe_get(containerapp_def, 'properties', 'circuitBreakerPolicy')
+        if circuit_breaker_policy and ('consecutiveErrors' in circuit_breaker_policy or 'intervalInSeconds' in circuit_breaker_policy or 'maxEjectionPercent' in circuit_breaker_policy):
+            circuit_breaker_policy['consecutiveErrors'] = circuit_breaker_policy.get('consecutiveErrors', DEFAULT_CONSECUTIVE_ERRORS)
+            circuit_breaker_policy['intervalInSeconds'] = circuit_breaker_policy.get('intervalInSeconds', DEFAULT_INTERVAL)
+            circuit_breaker_policy['maxEjectionPercent'] = circuit_breaker_policy.get('maxEjectionPercent', DEFAULT_MAX_EJECTION)
+            containerapp_def['properties']['circuitBreakerPolicy'] = circuit_breaker_policy
+
+        # HTTP Connection Pool
+        http_connection_pool = safe_get(containerapp_def, 'properties', 'httpConnectionPool')
+        if http_connection_pool and ('http1MaxPendingRequests' in http_connection_pool or 'http2MaxRequests' in http_connection_pool):
+            http_connection_pool['http1MaxPendingRequests'] = http_connection_pool.get('http1MaxPendingRequests', DEFAULT_HTTP1_MAX_PENDING_REQ)
+            http_connection_pool['http2MaxRequests'] = http_connection_pool.get('http2MaxRequests', DEFAULT_HTTP2_MAX_REQ)
+            containerapp_def['properties']['httpConnectionPool'] = http_connection_pool
+
+        return containerapp_def
+
+
+class ContainerAppResiliencyPreviewCreateDecorator(ContainerAppResiliencyDecorator):
+    def __init__(self, cmd: AzCliCommand, client: Any, raw_parameters: Dict, models: str):
+        super().__init__(cmd, client, raw_parameters, models)
+        self.containerapp_resiliency_def = ContainerAppsResiliencyModel
+
+    def create(self):
+        try:
+            r = self.client.create_or_update(
+                cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
+                name=self.get_argument_name(), container_app_name=self.get_argument_container_app_name(),
+                container_app_resiliency_envelope=self.containerapp_resiliency_def,
+                no_wait=self.get_argument_no_wait())
+            r = clean_null_values(r)
```

Comment:
May I ask why need to clean null values here? As other command will not clean the null values.

Reply:
This is because other values were being shown as `null` which is not required. We only want to show the set fields to the user.